sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAsyncClientBuilder.java

```diff
@@ -0,0 +1,152 @@
+import com.azure.security.attestation.implementation.AttestationClientImpl;
+import com.azure.security.attestation.implementation.AttestationClientImplBuilder;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Objects;
+
+/** A builder for creating a new instance of the AttestationClient type. */
+@ServiceClientBuilder(
+        serviceClients = {
+            AttestationAsyncClient.class,
+            SigningCertificatesAsyncClient.class
+        })
+public final class AttestationAsyncClientBuilder {
+    private final AttestationClientImplBuilder clientImplBuilder;
+    private final ClientLogger logger = new ClientLogger(AttestationAsyncClientBuilder.class);
+
+    /**
+     * Creates a new instance of the AttestationClientBuilder class.
+     */
+    public AttestationAsyncClientBuilder() {
+        clientImplBuilder = new AttestationClientImplBuilder();
+    }
+
+    /**
+     * Sets The attestation endpoint URI, for example https://mytenant.attest.azure.net.
+     *
+     * @param endpoint The endpoint to connect to.
+     * @return the AttestationClientBuilder.
+     */
+    public AttestationAsyncClientBuilder endpoint(String endpoint) {
+        Objects.requireNonNull(endpoint);
+        try {
+            new URL(endpoint);
+        } catch (MalformedURLException ex) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(ex));
+        }
+        clientImplBuilder.instanceUrl(endpoint);
+        return this;
+    }
+
+    /**
+     * Sets The HTTP pipeline to send requests through.
+     *
+     * @param pipeline the pipeline value.
+     * @return the AttestationClientBuilder.
+     */
+    public AttestationAsyncClientBuilder pipeline(HttpPipeline pipeline) {
+        Objects.requireNonNull(pipeline);
+        clientImplBuilder.pipeline(pipeline);
+        return this;
+    }
+
+    /**
+     * Sets The serializer to serialize an object into a string.
+     *
+     * @param serializerAdapter the serializerAdapter value.
+     * @return the AttestationClientBuilder.
+     */
+    public AttestationAsyncClientBuilder serializerAdapter(SerializerAdapter serializerAdapter) {
+        Objects.requireNonNull(serializerAdapter);
+        clientImplBuilder.serializerAdapter(serializerAdapter);
+        return this;
+    }
+
+    /**
+     * Sets The HTTP client used to send the request.
+     *
+     * @param httpClient the httpClient value.
+     * @return the AttestationClientBuilder.
+     */
+    public AttestationAsyncClientBuilder httpClient(HttpClient httpClient) {
+        clientImplBuilder.httpClient(httpClient);
+        return this;
+    }
+
+    /**
+     * Sets The configuration store that is used during construction of the service client.
+     *
+     * @param configuration the configuration value.
+     * @return the AttestationClientBuilder.
+     */
+    public AttestationAsyncClientBuilder configuration(Configuration configuration) {
```

Comment:
I do not see any credential support here - how does the service do authorization?

Reply:
The service requires a bearer token for most incoming requests.

This code is simply a clone of the generated client code - I took the generated builder and echoed all the methods on the generated builder. I have no idea why the autorest generated code didn't include credential support, but I assume that if autorest didn't think it was necessary that it wasn't a requirement.

Reply:
Credential support is not by default added because each service uses a different authentication scheme. 

Having said that I believe more recent versions of java autorest do provide bearerToken authentication support and you can use that to regenerate the builders.

Let me get the instructions for you if you would prefer to use that.

Reply:
So the changes are checked-in. If you use the latest version of autorest to generate your code - you will get it.

Reply:
For the other 3 languages, I added:

```
add-credentials: true
credential-scopes: 'https://attest.azure.net/.default'
```

Does this work for Java as well?

Reply:
I believe the argument is credential-types: tokencredential

Reply:
Ah, that makes a difference. Added.